### PR TITLE
ticdc: update some usage restrictions and configs 

### PR DESCRIPTION
**Pull request by amyangfei**
### What is changed, added or deleted? (Required)

- update some restrictions
- update dispatcher rule

### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [x]  v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Have version specific changes 
- [ ] Might cause conflicts

### CODE REVIEW

**Comment by amyangfei** [comment]
PTAL @TomShawn @leoppro

**Review by leoppro — COMMENTED** [review]
LGTM

File: ticdc/manage-ticdc.md [hunk 1]
```diff
@@ -583,7 +589,11 @@ worker-num = 16
 
 [sink]
 # 对于 MQ 类的 Sink，可以通过 dispatchers 配置 event 分发器
-# 支持 default、ts、rowid、table 四种分发器
+# 支持 default、ts、rowid、table 四种分发器，分发规则如下：
+# - default：有多个唯一索引（包括主键）时按照 table 模式分发；只有一个唯一索引（或主键）按照 rowid 模式分发；如果开启了 old value 特性，按照 table 分发
```
Comment (TomShawn) [inline ticdc/manage-ticdc.md:593]:
```suggestion
# - default：有多个唯一索引（包括主键）时按照 table 模式分发；只有一个唯一索引（或主键）按照 rowid 模式分发；如果开启了 old value 特性，按照 ts 模式分发
```

File: ticdc/manage-ticdc.md [hunk 2]
```diff
@@ -583,7 +589,11 @@ worker-num = 16
...
+# - ts：以行变更的 commitTs 做 hash 进行分发
+# - rowid：以选取的作为 HandleKey 的列名和列值做 hash 进行分发
```
Comment (TomShawn) [inline ticdc/manage-ticdc.md:595]:
```suggestion
# - rowid：以选作 HandleKey 的列名和列值做 hash 进行分发
```

File: ticdc/troubleshoot-ticdc.md [hunk 3]
```diff
@@ -62,6 +62,7 @@ cdc cli changefeed create --pd=http://10.0.10.25:2379 --start-ts=415238226621235
 > **注意：**
 >
 > TiCDC（4.0 发布版本）还不支持动态修改文件排序和内存排序。
+> 文件排序目前的处理能力有限，如果单表数据量过多导致文件排序也失败，可以修改 TiCDC 任务配置过滤掉这张表，通过其他备份恢复工具比如 br 恢复这张表之后再继续同步该表。
```
Comment (TomShawn) [inline ticdc/troubleshoot-ticdc.md:65]:
```suggestion
> + TiCDC（4.0 发布版本）还不支持动态修改文件排序和内存排序。
> + 目前文件排序功能的处理能力有限。如果单表数据量过多导致文件排序失败，可以修改 TiCDC 任务配置过滤掉这张表，通过其他备份恢复工具例如 BR 恢复这张表之后再继续同步该表。
```

File: ticdc/manage-ticdc.md [hunk 4]
```diff
@@ -583,7 +589,11 @@ worker-num = 16
 
 [sink]
 # 对于 MQ 类的 Sink，可以通过 dispatchers 配置 event 分发器
-# 支持 default、ts、rowid、table 四种分发器
+# 支持 default、ts、rowid、table 四种分发器，分发规则如下：
+# - default：有多个唯一索引（包括主键）时按照 table 模式分发；只有一个唯一索引（或主键）按照 rowid 模式分发；如果开启了 old value 特性，按照 table 分发
+# - ts：以行变更的 commitTs 做 hash 进行分发
+# - rowid：以所选的 HandleKey 列名和列值做 Hash 计算并进行分发
+# - table：以表的 schema 名和 table 名做 hash 进行分发
```
Comment (TomShawn) [inline ticdc/manage-ticdc.md:596]:
```suggestion
# - ts：以行变更的 commitTs 做 Hash 计算并进行分发
# - rowid：以所选的 HandleKey 列名和列值做 Hash 计算并进行分发
# - table：以表的 schema 名和 table 名做 Hash 计算进行分发
```

File: ticdc/manage-ticdc.md [hunk 5]
```diff
@@ -583,7 +589,11 @@ worker-num = 16
 
 [sink]
 # 对于 MQ 类的 Sink，可以通过 dispatchers 配置 event 分发器
-# 支持 default、ts、rowid、table 四种分发器
+# 支持 default、ts、rowid、table 四种分发器，分发规则如下：
+# - default：有多个唯一索引（包括主键）时按照 table 模式分发；只有一个唯一索引（或主键）按照 rowid 模式分发；如果开启了 old value 特性，按照 table 分发
+# - ts：以行变更的 commitTs 做 Hash 计算并进行分发
+# - rowid：以所选的 HandleKey 列名和列值做 Hash 计算并进行分发
+# - table：以表的 schema 名和 table 名做 Hash 计算并进行分发
```
Comment (TomShawn) [inline ticdc/manage-ticdc.md:596]:
```suggestion
# - ts：以行变更的 commitTs 做 Hash 计算并进行 event 分发
# - rowid：以所选的 HandleKey 列名和列值做 Hash 计算并进行 event 分发
# - table：以表的 schema 名和 table 名做 Hash 计算并进行 event 分发
```

**Review by TomShawn — APPROVED** [review]
LGTM

**Pull request merged**